File: 2373-largest-local-values-in-a-matrix/2373-largest-local-values-in-a-matrix.py
class Solution:
    def largestLocal(self, grid: List[List[int]]) -> List[List[int]]:
        maxLocal = [[0 for i in range(len(grid) - 2)] for j in range(len(grid) - 2)]

        for i in range(len(grid) - 2):
            for j in range(len(grid) - 2):
                for k in range(3):
                    for l in range(3):
                        maxLocal[i][j] = max(maxLocal[i][j], grid[i + k][j + l])

        return maxLocal

File: 2373-largest-local-values-in-a-matrix/2373-largest-local-values-in-a-matrix.java
class Solution {
    public int[][] largestLocal(int[][] grid) {
        int[][] maxLocal = new int[grid.length - 2][grid.length - 2];

        for (int i = 0; i < grid.length - 2; i++) {
            for (int j = 0; j < grid.length - 2; j++) {
                for (int k = 0; k < 3; k++) {
                    for (int l = 0; l < 3; l++) {
                        maxLocal[i][j] = Math.max(maxLocal[i][j], grid[i + k][j + l]);
                    }
                }
            }
        }

        return maxLocal;
    }
}